Build Windows wallet issue

<!-- This issue tracker is only for technical issues related to Bitcoin Core.

General bitcoin questions and/or support requests are best directed to the Bitcoin StackExchange at https://bitcoin.stackexchange.com.

For reporting security issues, please read instructions at https://bitcoincore.org/en/contact/.

If the node is "stuck" during sync or giving "block checksum mismatch" errors, please ensure your hardware is stable by running memtest and observe CPU temperature with a load-test tool such as linpack before creating an issue! -->

<!-- Describe the issue -->
<!--- What behavior did you expect? -->

<!--- What was the actual behavior (provide screenshots if the issue is GUI-related)? -->

<!--- How reliably can you reproduce the issue, what are the steps to do so? -->

<!-- What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)? -->

<!-- What type of machine are you observing the error on (OS/CPU and disk type)? -->

<!-- Any extra information that might be useful in the debugging process. -->
<!--- This is normally the contents of a `debug.log` or `config.log` file. Raw text or a link to a pastebin type site are preferred. -->

When I tried to build windows wallet on WSL as written on build-windows.md, I am getting these errors, and failure on this.

Please help me how to fix these issues.

>   OBJCXXLD qt/bitcoin-qt.exe
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x4e1): undefined reference to `SSL_accept'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x4f1): undefined reference to `SSL_clear'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x501): undefined reference to `SSL_CIPHER_description'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x511): undefined reference to `SSL_CIPHER_get_bits'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x521): undefined reference to `SSL_connect'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x531): undefined reference to `SSL_CTX_check_private_key'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x541): undefined reference to `SSL_CTX_ctrl'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x551): undefined reference to `SSL_CTX_free'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x561): undefined reference to `SSL_CTX_new'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x571): undefined reference to `SSL_CTX_set_cipher_list'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x581): undefined reference to `SSL_CTX_set_default_verify_paths'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x591): undefined reference to `SSL_CTX_set_verify'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x5a1): undefined reference to `SSL_CTX_set_verify_depth'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x5b1): undefined reference to `SSL_CTX_use_certificate'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x5c1): undefined reference to `SSL_CTX_use_certificate_file'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x5d1): undefined reference to `SSL_CTX_use_PrivateKey'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x5e1): undefined reference to `SSL_CTX_use_RSAPrivateKey'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x5f1): undefined reference to `SSL_CTX_use_PrivateKey_file'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x601): undefined reference to `SSL_CTX_get_cert_store'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x611): undefined reference to `SSL_free'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x621): undefined reference to `SSL_get_ciphers'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x631): undefined reference to `SSL_get_current_cipher'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x641): undefined reference to `SSL_version'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x651): undefined reference to `SSL_get_error'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x661): undefined reference to `SSL_get_peer_cert_chain'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x671): undefined reference to `SSL_get_peer_certificate'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x681): undefined reference to `SSL_get_verify_result'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x691): undefined reference to `SSL_library_init'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x6a1): undefined reference to `SSL_load_error_strings'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x6b1): undefined reference to `SSL_new'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x6c1): undefined reference to `SSL_ctrl'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x6d1): undefined reference to `SSL_read'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x6e1): undefined reference to `SSL_set_bio'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x6f1): undefined reference to `SSL_set_accept_state'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x701): undefined reference to `SSL_set_connect_state'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x711): undefined reference to `SSL_shutdown'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x721): undefined reference to `SSL_set_session'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x731): undefined reference to `SSL_SESSION_free'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x741): undefined reference to `SSL_get1_session'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x751): undefined reference to `SSL_get_session'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x761): undefined reference to `SSL_get_ex_new_index'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x771): undefined reference to `SSL_set_ex_data'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x781): undefined reference to `SSL_get_ex_data'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x791): undefined reference to `SSL_set_psk_client_callback'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x7a1): undefined reference to `SSL_set_psk_server_callback'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x7b1): undefined reference to `SSL_CTX_use_psk_identity_hint'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x7c1): undefined reference to `SSLv3_client_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x7d1): undefined reference to `SSLv23_client_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x7e1): undefined reference to `TLSv1_client_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x7f1): undefined reference to `TLSv1_1_client_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x801): undefined reference to `TLSv1_2_client_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x811): undefined reference to `SSLv3_server_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x821): undefined reference to `SSLv23_server_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x831): undefined reference to `TLSv1_server_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x841): undefined reference to `TLSv1_1_server_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x851): undefined reference to `TLSv1_2_server_method'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0x861): undefined reference to `SSL_write'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0xac1): undefined reference to `SSL_CTX_load_verify_locations'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0xaf1): undefined reference to `i2d_SSL_SESSION'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0xb01): undefined reference to `d2i_SSL_SESSION'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0xb11): undefined reference to `SSL_select_next_proto'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0xb21): undefined reference to `SSL_CTX_set_next_proto_select_cb'
> /home/god/bitcoin/depends/x86_64-w64-mingw32/share/../lib/libQt5Network.a(qsslsocket_openssl_symbols.o):qsslsocket_openssl_symbols.cpp:(.text+0xb31): undefined reference to `SSL_get0_next_proto_negotiated'
> collect2: error: ld returned 1 exit status
> Makefile:3927: recipe for target 'qt/bitcoin-qt.exe' failed
> make[2]: *** [qt/bitcoin-qt.exe] Error 1
> make[2]: Leaving directory '/home/god/bitcoin/src'
> Makefile:10249: recipe for target 'all-recursive' failed
> make[1]: *** [all-recursive] Error 1
> make[1]: Leaving directory '/home/god/bitcoin/src'
> Makefile:774: recipe for target 'all-recursive' failed
> make: *** [all-recursive] Error 1

--- a/src/Makefile.qt.include
+++ b/src/Makefile.qt.include
@@ -421,6 +421,10 @@ qt_bitcoin_qt_LDADD += $(LIBBITCOIN_CLI) $(LIBBITCOIN_COMMON) $(LIBBITCOIN_UTIL)
   $(EVENT_PTHREADS_LIBS) $(EVENT_LIBS)
 if ENABLE_BIP70
 qt_bitcoin_qt_LDADD += $(SSL_LIBS)
+else
+if TARGET_WINDOWS
+qt_bitcoin_qt_LDADD += $(SSL_LIBS)
+endif
 endif
 qt_bitcoin_qt_LDADD += $(CRYPTO_LIBS)
 qt_bitcoin_qt_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(QT_LDFLAGS) $(LIBTOOL_APP_LDFLAGS)

